File: ArrayProduct.java
import java.util.*;

class ArrayProduct {
  public int[] arrayOfProducts(int[] array) {
    
		int[] products=new int[array.length];
		
		int leftProduct=1;
		for(int i=0;i<array.length;i++){
			products[i]=leftProduct;
			leftProduct*=array[i];
		}
		
		int rightProduct=1;
		for(int i=array.length-1;i>=0;i--){
			products[i]*=rightProduct;
			rightProduct*=array[i];
		}
		
		return products;
    
  }
}

File: ArrayProduct.swift
class ArrayProduct {
	
  func arrayOfProducts(_ array: [Int]) -> [Int] {
    var products=[Int](repeating:1,count:array.count)
		
		var leftProduct=1
		for i in 0..<array.count{
			products[i]=leftProduct
			leftProduct*=array[i]
		}
		
		var rightProduct=1
		for i in stride(from:array.count-1, through:0, by:-1){
			products[i]*=rightProduct
			rightProduct*=array[i]
		}
		return products
  }
}
